addons/advanced_ballistics/functions/fnc_handleFired.sqf

```diff
@@ -0,0 +1,408 @@
+};
+
+_twistDirection = 1;
+if (isNumber(configFile >> "cfgWeapons" >> _weapon >> "ACE_twistDirection")) then {
+    _twistDirection = getNumber(configFile >> "cfgWeapons" >> _weapon >> "ACE_twistDirection");
+    if (_twistDirection != -1 && _twistDirection != 0 &&  _twistDirection != 1) then {
+        _twistDirection = 1;
+    };
+};
+
+_transonicStabilityCoef = 0.5;
+if (isNumber(configFile >> "cfgAmmo" >> _ammo >> "ACE_transonicStabilityCoef")) then {
+    _transonicStabilityCoef = getNumber(configFile >> "cfgAmmo" >> _ammo >> "ACE_transonicStabilityCoef");
+};
+
+_dragModel = 1;
+_ballisticCoefficients = [];
+_velocityBoundaries = [];
+_atmosphereModel = "ICAO";
+if (GVAR(AdvancedAirDragEnabled)) then {
+    if (isNumber(configFile >> "cfgAmmo" >> _ammo >> "ACE_dragModel")) then {
+        _dragModel = getNumber(configFile >> "cfgAmmo" >> _ammo >> "ACE_dragModel");
+        if (!(_dragModel in [1, 2, 5, 6, 7, 8])) then {
+            _dragModel = 1;
+        };
+    };
+    if (isArray(configFile >> "cfgAmmo" >> _ammo >> "ACE_ballisticCoefficients")) then {
+        _ballisticCoefficients = getArray(configFile >> "cfgAmmo" >> _ammo >> "ACE_ballisticCoefficients");
+    };
+    if (isArray(configFile >> "cfgAmmo" >> _ammo >> "ACE_velocityBoundaries")) then {
+        _velocityBoundaries = getArray(configFile >> "cfgAmmo" >> _ammo >> "ACE_velocityBoundaries");
+    };
+    if (isText(configFile >> "cfgAmmo" >> _ammo >> "ACE_standardAtmosphere")) then {
+        _atmosphereModel = getText(configFile >> "cfgAmmo" >> _ammo >> "ACE_standardAtmosphere");
+    };
+};
+
+_index = count GVAR(bulletDatabase);
+if (count GVAR(bulletDatabaseFreeIndices) > 0) then {
+    _index = GVAR(bulletDatabaseFreeIndices) select 0;
+    GVAR(bulletDatabaseFreeIndices) = GVAR(bulletDatabaseFreeIndices) - [_index];
+};
+
+#ifdef USE_ADVANCEDBALLISTICS_DLL
+    "ace_advanced_ballistics" callExtension format["new:%1:%2:%3:%4:%5:%6:%7:%8:%9:%10:%11:%12:%13:%14:%15:%16:%17:%18", _index, _airFriction, _ballisticCoefficients, _velocityBoundaries, _atmosphereModel, _dragModel, _stabilityFactor, _twistDirection, _muzzleVelocity, _transonicStabilityCoef, getPosASL _bullet, EGVAR(weather,Latitude), EGVAR(weather,currentTemperature), EGVAR(weather,Altitude), EGVAR(weather,currentHumidity), overcast, floor(time), time - floor(time)];
+    GVAR(bulletDatabase) set[_index, [_bullet, _caliber, _bulletTraceVisible, _index]];
+
+    if ((GVAR(bulletDatabaseOccupiedIndices) pushBack _index) == 0) then {
+        [{
+            private ["_bulletDatabaseEntry", "_index", "_bullet", "_caliber", "_bulletTraceVisible", "_bulletVelocity", "_bulletPosition"];
+
+            {
+                _bulletDatabaseEntry = (GVAR(bulletDatabase) select _x);
+                _bullet = _bulletDatabaseEntry select 0;
+                _index  = _bulletDatabaseEntry select 3;
+                if (!alive _bullet) then {
```

Comment:
`_index` is only used after this if statement. The assignement should be after this if statement, as that is when we need it. Assignment of the bullet variable is also not necessary here.

```
if (!alive (_bulletDatabaseEntry select 0) /* bullet */) then {
   _index = _bulletDatabaseEntry select 3;
...
```